lib/swaps/Swaps.ts

```diff
@@ -1,67 +1,68 @@
-import { SwapDealRole } from '../types/enums';
+import { SwapDealRole, SwapDealState, SwapDealStatus } from '../types/enums';
 import Peer from '../p2p/Peer';
-import uuidv1 from 'uuid/v1';
 import * as packets from '../p2p/packets/types';
-import { randomBytes, createHash } from 'crypto';
+import { createHash, randomBytes } from 'crypto';
 import Logger from '../Logger';
 import * as lndrpc from '../proto/lndrpc_pb';
 import LndClient from '../lndclient/LndClient';
 import Pool from '../p2p/Pool';
 import { EventEmitter } from 'events';
-import { StampedPeerOrder, StampedOwnOrder } from '../types/orders';
+import { StampedOwnOrder, StampedPeerOrder } from '../types/orders';
+import assert from 'assert';
 
 type SwapDeal = {
   /** The role of the local node in the swap. */
   myRole: SwapDealRole;
-  /** The xud node pub key of the counterparty to this swap deal. */
-  peerPubKey: string;
+  /** The most updated deal status */
+  status: SwapDealStatus;
+  /** The most updated deal state. Works together with status. */
+  state: SwapDealState;
```

Comment:
Having both `state` and `status` is confusing to me, can you combine the logic into a single field or at least explain in comments specifically what they represent and how they are used differently? It seems right now state is only being used for setting the error state - can we just have a `SwapError` status instead?

Reply:
State and Status are not the same. 
The idea is that know that the deal is in error state and at which phase it was while having the error.  If we set SwapError status we lose the deal status at the time of the error. 
I updated the comment.

Reply:
OK that's what I was thinking too and I see the value in preserving where we were in the swap flow (the status), but I still think the naming is confusing here given that state and status are synonyms.  But will elaborate in the other comment regarding the `setDealState` method.